### PR TITLE
Feature/inpro 474 switch gworkspace provider to official

### PR DESCRIPTION
**Pull request by jazzlyn**
please review.

### CODE REVIEW

File: examples/example-provider-authentication/main.tf [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
+terraform {
+  required_providers {
+    googleworkspace = {
+      source  = "hashicorp/googleworkspace"
+      version = ""
+    }
+  }
+}
+
+provider "googleworkspace" {
```
Comment (tyriis) [inline examples/example-provider-authentication/main.tf:15]: for me it looks like a bad practice to set the credentials in the repo vars as it increase the chance to commit them by accident. Also for ci systems it is harder to configure a temporary file then to use then ENV values.

There is an ENV configuration to read the credentials. with tools like [direnv](https://direnv.net/) it is easy to set it up for local development aswel as for ci systems.

File: examples/example-provider-authentication/terraform.tfvars.example [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# -------------------------------------------------------------------------------------------
```
Comment (tyriis) [inline examples/example-provider-authentication/terraform.tfvars.example:1]: as mentioned in the credential provider args

File: examples/example-with-google-buckets/main.tf [hunk 3]
```diff
@@ -1,24 +1,25 @@
...
 # ---------------------------------------------------------------------------------------------------------------------
 
 terraform {
   required_providers {
-    gsuite = {
-      source  = "DeviaVir/gsuite"
-      version = "0.1.58"
+    googleworkspace = {
+      source  = "hashicorp/googleworkspace"
+      version = ""
```
Comment (tyriis) [inline examples/example-with-google-buckets/main.tf:11]: pin to versions so renovate will update examples for you

File: examples/example-with-local-files/main.tf [hunk 4]
```diff
@@ -6,16 +6,16 @@
 
 terraform {
   required_providers {
-    gsuite = {
-      source  = "DeviaVir/gsuite"
-      version = "0.1.58"
+    googleworkspace = {
+      source  = "hashicorp/googleworkspace"
+      version = ""
```
Comment (tyriis) [inline examples/example-with-local-files/main.tf:11]: pin to versions so renovate will update them for you

File: examples/example-with-local-files/main.tf [hunk 5]
```diff
@@ -37,10 +37,10 @@ locals {
 # ---------------------------------------------------------------------------------------------------------------------
 
 module "user-group-management" {
-  source  = "strg-at/user-group-management/gsuite"
-  version = "0.1.0"
+  source  = "strg-at/user-group-management/googleworkspace"
+  version = ""
   providers = {
-    gsuite = gsuite
+    googleworkspace = googleworkspace
```
Comment (tyriis) [inline examples/example-with-local-files/main.tf:43]: is this still required?

File: examples/example-yaml-files/groups.yaml [hunk 6]
```diff
@@ -3,21 +3,21 @@
 # ---------------------------------------------------------------------------------------------------------------------
 
 one-group:
-  email: 'group@example.io'
-  name: 'Example group'
-  settings: 'anyone-mail-allowed'
+  email: "group@example.io"
```
Comment (tyriis) [inline examples/example-yaml-files/groups.yaml:6]: would not use `"` in yaml if you don't need them

**Review by tyriis — COMMENTED** [review]
Overall it looks okay, please check the auth config part would strongly suggest to change it.

File: examples/example-provider-authentication/terraform.tfvars.example (hunk 2 above)
Comment (jazzlyn) [reply examples/example-provider-authentication/terraform.tfvars.example:1]: will consider this for the data repo, I added with-tfvars" to the example folder name to be clear with the example that is shown here.

File: examples/example-with-local-files/main.tf (hunk 5 above)
Comment (jazzlyn) [reply examples/example-with-local-files/main.tf:43]: yes

**Review by tyriis — APPROVED** [review]

File: examples/example-yaml-files/group_settings.yaml [hunk 7]
```diff
@@ -1,34 +1,28 @@
...
-  include_in_global_address_list: null
-  spam_moderation_level: null
-  who_can_contact_owner: null
-  who_can_discover_group: null
-  who_can_join: null
-  who_can_leave_group: null
-  who_can_post_message: null
-  who_can_view_group: null
-  who_can_view_membership: null
+# set values to null
```
[the remark this answers is not in the document]
Comment (jazzlyn) [reply examples/example-yaml-files/group_settings.yaml:28]: added information as comment

**Pull request merged**